crates/bevy_render/src/mesh/primitives/dim2.rs

```diff
@@ -193,6 +195,124 @@ impl From<EllipseMeshBuilder> for Mesh {
     }
 }
 
+/// A builder for creating a [`Mesh`] with an [`Annulus`] shape.
+pub struct AnnulusMeshBuilder {
+    /// The [`Annulus`] shape.
+    pub annulus: Annulus,
+
+    /// The number of vertices used in constructing each concentric circle of the annulus mesh.
+    /// The default is `32`.
+    pub resolution: usize,
+}
+
+impl Default for AnnulusMeshBuilder {
+    fn default() -> Self {
+        Self {
+            annulus: Annulus::default(),
+            resolution: 32,
+        }
+    }
+}
+
+impl AnnulusMeshBuilder {
+    /// Create an [`AnnulusMeshBuilder`] with the given inner radius, outer radius, and angular vertex count.
+    #[inline]
+    pub fn new(inner_radius: f32, outer_radius: f32, resolution: usize) -> Self {
+        Self {
+            annulus: Annulus::new(inner_radius, outer_radius),
+            resolution,
+        }
+    }
+
+    /// Sets the number of vertices used in constructing the concentric circles of the annulus mesh.
+    #[inline]
+    pub fn resolution(mut self, resolution: usize) -> Self {
+        self.resolution = resolution;
+        self
+    }
+
+    /// Builds a [`Mesh`] based on the configuration in `self`.
+    pub fn build(&self) -> Mesh {
+        let inner_radius = self.annulus.inner_circle.radius;
+        let outer_radius = self.annulus.outer_circle.radius;
+
+        let num_vertices = (self.resolution + 1) * 2;
+        let mut indices = Vec::with_capacity(num_vertices * 3);
```

Comment:
I think this is slightly larger than necessary? The loop adds 2 sets of 3 indices `self.resolution` times, not `self.resolution + 1` times
```suggestion
        let mut indices = Vec::with_capacity(self.resolution * 6);
```

Reply:
Nice catch!